elixir/apps/api/lib/api/gateway/channel.ex

```diff
@@ -130,16 +136,43 @@ defmodule API.Gateway.Channel do
     end
   end
 
-  def handle_info({:reject_access, client_id, resource_id}, socket) do
+  # Resource is updated, eg. traffic filters are changed
+  def handle_info({:update_resource, resource_id}, socket) do
+    OpenTelemetry.Ctx.attach(socket.assigns.opentelemetry_ctx)
+    OpenTelemetry.Tracer.set_current_span(socket.assigns.opentelemetry_span_ctx)
+
+    OpenTelemetry.Tracer.with_span "gateway.resource_updated",
+      attributes: %{resource_id: resource_id} do
+      resource = Resources.fetch_resource_by_id!(resource_id)
+      push(socket, "resource_updated", Views.Resource.render(resource))
+      {:noreply, socket}
+    end
+  end
+
+  # This event is ignored because we will receive a reject_access message from
+  # the Flows which will trigger a reject_access event
+  def handle_info({:delete_resource, resource_id}, socket) do
+    :ok = Resources.unsubscribe_from_events_for_resource(resource_id)
+    {:noreply, socket}
+  end
+
+  # Flows context broadcasts this message when flow is expired,
+  # which happens when policy, resource, actor, group, identity or provider were
+  # disabled or deleted
+  def handle_info({:expire_flow, flow_id, client_id, resource_id}, socket) do
     OpenTelemetry.Ctx.attach(socket.assigns.opentelemetry_ctx)
     OpenTelemetry.Tracer.set_current_span(socket.assigns.opentelemetry_span_ctx)
 
     OpenTelemetry.Tracer.with_span "gateway.reject_access",
       attributes: %{
+        flow_id: flow_id,
         client_id: client_id,
         resource_id: resource_id
       } do
+      :ok = Flows.unsubscribe_to_flow_expiration_events(flow_id)
+
       push(socket, "reject_access", %{
+        flow_id: flow_id,
```

Comment:
@conectado `reject_access` also had `flow_id`, you can ignore it for now.

Reply:
`reject_access` would only remove access for the combination of (`client_id`, `resource_id`)?

Reply:
That should be enough for now, yes. But we can also just use Flow ID instead.